manager/src/main/java/com/redhat/service/bridge/manager/connectors/ConnectorsServiceImpl.java

```diff
@@ -0,0 +1,176 @@
+import com.redhat.service.bridge.manager.models.ConnectorEntity;
+import com.redhat.service.bridge.manager.models.Processor;
+
+@ApplicationScoped
+public class ConnectorsServiceImpl implements ConnectorsService {
+
+    @Inject
+    ObjectMapper mapper;
+
+    @Inject
+    ConnectorsApi connectorsApi;
+
+    @Inject
+    ConnectorDAO connectorDAO;
+
+    @Inject
+    ProcessorDAO processorDAO;
+
+    @ConfigProperty(name = "mc.cluster.id")
+    String mcClusterId;
+
+    @ConfigProperty(name = "kafka.bootstrap.servers")
+    String kafkaBootstrapServer;
+
+    @ConfigProperty(name = "kafka.client.id")
+    String serviceAccountId;
+
+    @ConfigProperty(name = "kafka.client.secret")
+    String serviceAccountSecret;
+
+    @Override
+    @Transactional(Transactional.TxType.MANDATORY)
+    public Optional<String> createConnectorIfNeeded(ProcessorRequest processorRequest,
+            BaseAction resolvedAction,
+            Processor processor) {
+
+        BaseAction action = processorRequest.getAction();
+
+        if (!ConnectorsAction.TYPE.equals(action.getType())) {
+            return Optional.empty();
+        }
+
+        Map<String, String> actionParameters = action.getParameters();
+        String connectorType = actionParameters.get("connectorType");
+        String newConnectorName = actionParameters.get("connectorName");
+
+        JsonNode connectorPayload = parseConnectorPayloadString(actionParameters);
+
+        // Assume connectors actions are transformed to send-to-kafka
+        String kafkaTopicName = resolvedAction.getParameters().get(KafkaTopicAction.TOPIC_PARAM);
+        if (kafkaTopicName == null) {
+            throw new RuntimeException("Need a kafka topic to create a connector");
+        }
+        updateKafkaTopicInConnectorPayload(connectorPayload, kafkaTopicName);
+
+        persistConnector(processor, newConnectorName, connectorPayload);
+
+        try {
+            Connector connector = callConnectorService(connectorType, connectorPayload, newConnectorName);
+        } catch (Exception e) {
+            throw new RuntimeException(e); // TODO-MC error handling
+        }
+
+        return Optional.of(newConnectorName);
+    }
+
+    private void updateKafkaTopicInConnectorPayload(JsonNode connectorPayload, String kafkaTopicName) {
+        // Assume the Connector action is always transformed to a send-to-kafka Action
+        String kafkaTopic = kafkaTopicName;
+        ObjectNode kafka = connectorPayload.with("kafka");
+        kafka.put("topic", kafkaTopic);
+    }
+
+    private void persistConnector(Processor processor, String newConnectorName, JsonNode connectorPayload) {
+        ConnectorEntity newConnectorEntity = new ConnectorEntity();
+
+        newConnectorEntity.setName(newConnectorName);
+        newConnectorEntity.setStatus(ConnectorStatus.REQUESTED);
+        newConnectorEntity.setSubmittedAt(ZonedDateTime.now());
+        newConnectorEntity.setPublishedAt(ZonedDateTime.now());
```

Comment:
I assume this should be set once the Connector is fully deployed?

Reply:
https://issues.redhat.com/browse/MGDOBR-142